network/p2p/handler.go

```diff
@@ -33,13 +30,14 @@ type Handler interface {
 		gossipBytes []byte,
 	)
 	// AppRequest is called when handling an AppRequest message.
-	// Returns the bytes for the response corresponding to [requestBytes]
+	// Sends a response with the response corresponding to [requestBytes] or
+	// an application-defined error.
 	AppRequest(
 		ctx context.Context,
 		nodeID ids.NodeID,
 		deadline time.Time,
 		requestBytes []byte,
-	) ([]byte, error)
+	) ([]byte, *common.AppError)
```

Comment:
should we do returns an error code and error (which can be used to get the string?) so like:
```suggestion
	) ([]byte, int, error)
```
or we can make the int a named type to signal some codes (instead of the *common.AppError structs)

Reply:
I used the `AppError` struct since it implements the `Error` interface which is handy because the code handling the response can handle the error using `errors.Is`/other go code that handle the error interface.